clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4610,6 +4610,18 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
       CmdArgs.push_back(
           Args.MakeArgString(Twine("-fsycl-unique-prefix=") + UniqueID));
 
+    // Disable parallel for range-rounding for anything involving FPGA
+    auto SYCLTCRange = C.getOffloadToolChains<Action::OFK_SYCL>();
+    bool HasFPGA = false;
+    for (auto TI = SYCLTCRange.first, TE = SYCLTCRange.second; TI != TE; ++TI)
+      if (TI->second->getTriple().getSubArch() ==
+          llvm::Triple::SPIRSubArch_fpga) {
+        HasFPGA = true;
+        break;
+      }
+    if (HasFPGA)
```

Comment:
I am pretty sure this can be replaced by a call to `llvm::any_of()` instead of doing the loop manually (I'm not quite certain of the lambda signature for the predicate though as I'm not familiar with whatever the offload toolchain iterator is vending).

Reply:
I am familiar to this Driver area. This was added by @mdtoguchi. I have created https://github.com/intel/llvm/issues/3784 for this. @AaronBallman, can we do this in a separate PR?

Reply:
I'm fine handling it in a separate PR, it shouldn't be overly hard (it's an NFC change).

Reply:
Thanks @AaronBallman. I have assigned the issue to myself and i will take a look at this after this PR lands.